lib/sequenceserver/blast/job.rb

```diff
@@ -50,6 +60,19 @@ def command
                      " -query '#{qfile}' #{options}"
       end
 
+      def ncharacters_total
+        databases.map(&:ncharacters).map(&:to_i).reduce(:+)
+      end
+
+      def query_size
+        size = 0
+        IO.foreach(@qfile) do |line|
+          next if line[0] == '>'
+          size += line.strip.length
+        end
+        size
+      end
+
```

Comment:
Can we make the two methods private and call them `calculate_databases_ncharacters_total` and `calculate_query_size`.

Reply:
One question. Isn't it bad practice to test private methods? I know that I did it in one part as well but I just got a little deeper. But if you think that is ok for this situation, I will get this implemented.

Reply:
But you don't need to test/call the private methods in your test cases. You have called them already in the constructor and saved the return value in instance variables that are accessible through instance attr_accessors.